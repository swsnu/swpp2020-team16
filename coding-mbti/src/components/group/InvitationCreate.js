import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

import { useAlert } from 'react-alert';

export default function InvitationCreate(props) {
  const alert = useAlert();

  const { createInvitation, groupId, isManager } = props;

  const [open, setOpen] = React.useState(false);
  const [coderId, setCoderId] = React.useState('');

  const handleClickOpen = () => {
    if (!isManager) {
      alert.show('Only manager can invite coders.');
      return;
    }
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const createInvitationWithId = async () => {
    await createInvitation({ group_id: groupId, receiver: coderId });
    alert.show(`invitation sent to the coder [${coderId}]`);
    handleClose();
  };

  return (
    <>
      <Button
        variant="contained"
        size="large"
        color="primary"
        onClick={handleClickOpen}
      >
        Invite Coder!
      </Button>
      <Dialog open={open} onClose={handleClose} aria-labelledby="form-dialog-title">
        <DialogTitle id="form-dialog-title">Invite coder</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please enter inviting coder id
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="name"
            label="coder id"
            fullWidth
            onChange={e => setCoderId(e.target.value)}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            Cancel
          </Button>
          <Button id="InvitationStart" onClick={createInvitationWithId} color="primary">
            Invite
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
}

InvitationCreate.propTypes = {
  createInvitation: PropTypes.func.isRequired,
  groupId: PropTypes.string.isRequired,
  isManager: PropTypes.bool.isRequired,
};
