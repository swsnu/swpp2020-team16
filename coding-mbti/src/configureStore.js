import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  REGISTER,
  PAUSE,
  PERSIST,
  PURGE,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import reducer from './feature';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

export default () => {
  const store = configureStore({
    reducer: persistedReducer,
    middleware: getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
  });
  const persistor = persistStore(store);
  return { store, persistor };
};
