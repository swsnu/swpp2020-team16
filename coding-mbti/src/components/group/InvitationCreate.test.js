import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { ThemeProvider, createMuiTheme } from '@material-ui/core';
import { transitions, positions, Provider as AlertProvider } from 'react-alert';
import AlertTemplate from 'react-alert-template-basic';
import configureStore from '../../configureStore';
import InvitationCreate from './InvitationCreate';

const theme = createMuiTheme();

jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        push: jest.fn(),
    }),
}));

describe('GroupDelete', () => {
    it('render', () => {
        const { store } = configureStore();
        const options = {
            // you can also just use 'bottom center'
            position: positions.BOTTOM_CENTER,
            timeout: 5000,
            offset: '30px',
            // you can also just use 'scale'
            transition: transitions.SCALE,
        };

        const deleteGroup = jest.fn();

        const wrapper = mount(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <AlertProvider template={AlertTemplate} {...options}>
                        <InvitationCreate isManager createInvitation={deleteGroup} />
                    </AlertProvider>
                </ThemeProvider>
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
        const startButton = wrapper.find('button').first();
        startButton.simulate('click');

        const deleteButton = wrapper.find('#InvitationStart').first();
        deleteButton.simulate('click');
        expect(deleteGroup).toHaveBeenCalled();

        expect(wrapper).toMatchSnapshot();
    });
});
