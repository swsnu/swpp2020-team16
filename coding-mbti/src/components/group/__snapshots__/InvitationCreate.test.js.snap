// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupDelete render 1`] = `ReactWrapper {}`;

exports[`GroupDelete render 2`] = `ReactWrapper {}`;
